mlrun/platforms/other.py

```diff
@@ -165,3 +166,75 @@ def _mount_hostpath(task):
         )
 
     return _mount_hostpath
+
+
+def mount_s3(
+    aws_access_key="",
+    aws_secret_key="",
+    secret_name=None,
+    endpoint_url=None,
+    aws_region=None,
+):
+    """Modifier function to add s3 env vars or secrets to container
+
+    :param aws_access_key: AWS_ACCESS_KEY_ID value
+    :param aws_secret_key: AWS_SECRET_ACCESS_KEY value
+    :param endpoint_url: s3 endpoint address (for non AWS s3)
+    :param secret_name: kubernetes secret name (storing the access/secret keys)
+    :param aws_region: amazon region
+    :return:
+    """
+
+    if secret_name and (aws_access_key or aws_secret_key):
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "can use k8s_secret for credentials or specify them (aws_access_key, aws_secret_key) not both"
+        )
+
+    def _use_s3_cred(container_op):
+        from os import environ
+
+        from kubernetes import client as k8s_client
+
+        _access_key = aws_access_key or environ.get("AWS_ACCESS_KEY_ID")
+        _secret_key = aws_secret_key or environ.get("AWS_SECRET_ACCESS_KEY")
+        _endpoint_url = endpoint_url or environ.get("S3_ENDPOINT_URL")
+
+        container = container_op.container
+        if _endpoint_url:
+            container.add_env_variable(
+                k8s_client.V1EnvVar(name="S3_ENDPOINT_URL", value=endpoint_url)
+            )
+        if aws_region:
+            container.add_env_variable(
+                k8s_client.V1EnvVar(name="AWS_REGION", value=aws_region)
+            )
+
+        if secret_name:
+            container.add_env_variable(
+                k8s_client.V1EnvVar(
+                    name="AWS_ACCESS_KEY_ID",
+                    value_from=k8s_client.V1EnvVarSource(
+                        secret_key_ref=k8s_client.V1SecretKeySelector(
+                            name=secret_name, key="AWS_ACCESS_KEY_ID"
+                        )
+                    ),
+                )
+            ).add_env_variable(
+                k8s_client.V1EnvVar(
+                    name="AWS_SECRET_ACCESS_KEY",
+                    value_from=k8s_client.V1EnvVarSource(
+                        secret_key_ref=k8s_client.V1SecretKeySelector(
+                            name=secret_name, key="AWS_SECRET_ACCESS_KEY"
+                        )
+                    ),
+                )
+            )
+
+        else:
+            return container_op.add_env_variable(
+                k8s_client.V1EnvVar(name="AWS_ACCESS_KEY_ID", value=_access_key)
```

Comment:
I'm not comfortable with enabling this non-secure flow which will expose AWS creds in the function spec, as well as in the UI/sdk. At a minimum this should show a visible warning that notifies user about the potential risks (also in the docstring). 
I was thinking that this could be a utility function, such that if the user provides visible creds, it will automatically plant them (well, probably just the `AWS_SECRET_ACCESS_KEY`) in a project secret, saving the user the need to do that.

Reply:
we have the same in mount_v3io, i'm ok with adding a warning, and there is a way to use `secret_name`, we shouldn't make modifiers to complex. AFAIK the S3 datastore driver is already looking into the project secret

Reply:
added log warning